inginious/frontend/templates/task.html

```diff
@@ -75,7 +75,42 @@ <h3>$:_("Information")</h3>
                         $:_("{nb_submissions} submissions").format(nb_submissions=submission_limit["amount"])
             </td>
         </tr>
+        $ tags = ""
+        $for tag in task.get_tags()[2]:
+            $if tag.is_visible_for_student() or user_manager.has_staff_rights_on_course(course):
+                $ tags += (_(tag.get_name()) + ", ")
+        $ tags = tags[:-2]
+        $if tags != "":
+        <tr>
+            <td>Organisational Tags</td>
+            <td>$tags</td>
+        </tr>
     </table>
+     
+    <!-- Tags -->
+    $ number_of_visible_tags = 0
+    $ number_of_visible_antitag = 0
+    $for tag in task.get_tags()[0]:
+        $if tag.is_visible_for_student() or user_manager.has_staff_rights_on_course(course):
+            $ number_of_visible_tags += 1
+    $for tag in task.get_tags()[1]:
+        $if tag.is_visible_for_student() or user_manager.has_staff_rights_on_course(course):
+            $ number_of_visible_antitag += 1
+
+    $# Auto tags may appear for admins. So we force the h3 title even if there is not predefined tags
+    $# We do not show the tag header for students if they can not get tags
+    $if (number_of_visible_tags > 0) or (number_of_visible_antitag > 0) or user_manager.has_staff_rights_on_course(course):
```

Comment:
Using list comprehension and filtering to form new ``visible_tags`` and ``visible_antitags`` lists would probably avoid iterating several times here.

Something like : 

    visible_tags = [tag for tag in task.get_tags()[0] if tag.is_visible_for_student() or user_manager.has_staff_rights_on_course(course)]

And then ``if len(visible_tags)`` and ``for tag in visible_tags:``.